pyximport/pyximport.py

```diff
@@ -300,19 +301,31 @@ def find_module(self, fullname, package_path=None):
             paths = package_path
         else:
             paths = sys.path
-        join_path = os.path.join
-        is_file = os.path.isfile
-        is_abs = os.path.isabs
-        abspath = os.path.abspath
-        #is_dir = os.path.isdir
-        sep = os.path.sep
+
         for path in paths:
             if not path:
                 path = os.getcwd()
-            elif not is_abs(path):
-                path = abspath(path)
-            if is_file(path+sep+pyx_module_name):
-                return PyxLoader(fullname, join_path(path, pyx_module_name),
+            elif not os.path.isabs(path):
+                path = os.path.abspath(path)
+
+            try:
+                zi = zipimporter(path)
+                data = zi.get_data(pyx_module_name)
+            except (ZipImportError, IOError):
+                pyx_module_path = os.path.join(path, pyx_module_name)
```

Comment:
Wouldn't it be more efficient to look up the path first, and only failing that, try to zip-import it? This doesn't seem like a feature that all that many people would use, especially not during development.